### PR TITLE
Make this provider compatible with other architecture/GPU

### PR DESCRIPTION
**Pull request by petitfou**
KIVU-104

### CODE REVIEW

**Review by stolk — CHANGES_REQUESTED** [review]
Still needs a lot more work, it is too fragile:

```
Traceback (most recent call last):
  File "/tmp/nest-7uk9gjzm.6d7465c810e2f88d84d99eae2992f2463dbe401c2d3d7ab73c077d8a1846fa47/tdb", line 16, in <module>
    from tinydb import TinyDB, Query, where
ModuleNotFoundError: No module named 'tinydb'
```

and 

```
Traceback (most recent call last):
  File "/tmp/nest-kgprig99.80f7a2c9fe46ea62c533b6035098863a55ce829a36a413055fd31fad02eee6b0/tdb.py", line 26, in <module>
    import filelock
ModuleNotFoundError: No module named 'filelock'
Traceback (most recent call last):
  File "/tmp/nest-kgprig99.80f7a2c9fe46ea62c533b6035098863a55ce829a36a413055fd31fad02eee6b0/tdb.py", line 26, in <module>
    import filelock
ModuleNotFoundError: No module named 'filelock'
```

I think this one is because checkbox got confused? I have both radeon and intel gpu, but I am using the latter:

```
  File "/tmp/nest-5vak0hqn.39fcc67951e4c03b42f2ddeff46a784d572b25f5c718800cb22546bbfdd9aa4c/gpu-load.py", line 72, in load_radeon
    subprocess.check_call(f'radeontop -d {fname} --dump-interval=1 --limit={timeout}', stderr=subprocess.STDOUT, stdout=subprocess.DEVNULL, shell=True)
  File "/snap/checkbox22/current/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'radeontop -d gpu-load.data --dump-interval=1 --limit=10' returned non-zero exit status 1.
```

**Comment by petitfou** [comment]
> Still needs a lot more work, it is too fragile:
> 
> ```
> Traceback (most recent call last):
>   File "/tmp/nest-7uk9gjzm.6d7465c810e2f88d84d99eae2992f2463dbe401c2d3d7ab73c077d8a1846fa47/tdb", line 16, in <module>
>     from tinydb import TinyDB, Query, where
> ModuleNotFoundError: No module named 'tinydb'
> ```
> 
> and
> 
> ```
> Traceback (most recent call last):
>   File "/tmp/nest-kgprig99.80f7a2c9fe46ea62c533b6035098863a55ce829a36a413055fd31fad02eee6b0/tdb.py", line 26, in <module>
>     import filelock
> ModuleNotFoundError: No module named 'filelock'
> Traceback (most recent call last):
>   File "/tmp/nest-kgprig99.80f7a2c9fe46ea62c533b6035098863a55ce829a36a413055fd31fad02eee6b0/tdb.py", line 26, in <module>
>     import filelock
> ModuleNotFoundError: No module named 'filelock'
> ```
> 
> I think this one is because checkbox got confused? I have both radeon and intel gpu, but I am using the latter:
> 
> ```
>   File "/tmp/nest-5vak0hqn.39fcc67951e4c03b42f2ddeff46a784d572b25f5c718800cb22546bbfdd9aa4c/gpu-load.py", line 72, in load_radeon
>     subprocess.check_call(f'radeontop -d {fname} --dump-interval=1 --limit={timeout}', stderr=subprocess.STDOUT, stdout=subprocess.DEVNULL, shell=True)
>   File "/snap/checkbox22/current/usr/lib/python3.10/subprocess.py", line 369, in check_call
>     raise CalledProcessError(retcode, cmd)
> subprocess.CalledProcessError: Command 'radeontop -d gpu-load.data --dump-interval=1 --limit=10' returned non-zero exit status 1.
> ```

That because the provider requires : python3-tinydb & python3-filelock , could you please install them and retry ?

**Comment by stolk** [comment]
The tinydb error went away but the filelock error remains:
```
Traceback (most recent call last):
  File "/tmp/nest-9lbppmy8.eeafcce570392a0c71e2305f72f86035571a6066d1a8cb6aad4d6ceeb60e107e/tdb.py", line 26, in <module>
    import filelock
ModuleNotFoundError: No module named 'filelock'
Traceback (most recent call last):
  File "/tmp/nest-9lbppmy8.eeafcce570392a0c71e2305f72f86035571a6066d1a8cb6aad4d6ceeb60e107e/tdb.py", line 26, in <module>
    import filelock
ModuleNotFoundError: No module named 'filelock'
```

filelock is installed on DUT:

```
$ dpkg --list | grep python3-filelock
ii  python3-filelock                               3.7.1-1                                    all          platform independent file locking module
```

File: checkbox-provider-kivu/bin/gpu-load.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
```
Comment (stolk) [inline checkbox-provider-kivu/bin/gpu-load.py:1]: Maybe this needs to be `#!/usr/bin/python3` instead?

Comment (petitfou) [reply checkbox-provider-kivu/bin/gpu-load.py:1]: I added these two dependencies in the snapcraft.yaml file so they will be bundled with the snap

**Pull request merged**